File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/Specifications/Commands/SelectItemCommand.js
SIE.defineCommand('SIE.Web.MES.TaskManagement.Specifications.Commands.SelectItemCommand', {
    extend: 'SIE.cmd.LookupCommandBase',
    userConfig: {
        dataParams: { specKeyPrototyName: 'ProductId', targetClassName: 'SIE.Items.Item' },
    },
    meta: { text: "选择", group: "edit", iconCls: "icon-PlaylistCheck icon-blue" },
    save: function (win) {
        /// <summary>
        /// 保存选择的操作列表。
        /// </summary>
        var me = this;
        /* post数据结构*/
        var indata = {};
        /* post数据结构*/
        var selections = this._targetView.getSelection();
        if (selections && selections.length > 0) {
            var operationDatas = [];
            SIE.each(selections, function (item) {
                var productId = item.getId();
                if (me._sourceViewSelectItems.indexOf(productId) === -1) {
                    var productSpecification = { ProductId: productId };
                    operationDatas.push(productSpecification);
                }
            });
            indata = operationDatas;
            me._targetView.execute({
                data: indata,
                success: function (res) {
                    win.close();  //关闭模态窗口
                    me._ownerView.loadData(true); //重载视图数据
                }
            }, me._ownerView);
        }
        else {
            SIE.Msg.showWarning('没有可提交的数据'.t());
        }
    }
    // end 
});

File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/Specifications/Commands/SelectItemCommand.cs
using SIE.Domain;
using SIE.MES.TaskManagement.Specifications;
using SIE.Web.Command;
using System;
using System.Collections.Generic;

namespace SIE.Web.MES.TaskManagement.Specifications.Commands
{
    /// <summary>
    /// 产品规格件物料选择
    /// </summary>
    public class SelectItemCommand : ViewCommand
    {
        /// <summary>
        /// 执行选择
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var meta = ClientEntities.Find("SIE.MES.TaskManagement.Specifications.ProductSpecification");
            var savedData = RF.Find(meta.EntityType).NewList();
            var productSpecifications = args.Data.ToJsonObject<List<ProductSpecification>>();
            Check.NotNullOrEmpty(productSpecifications, nameof(productSpecifications));
            if (null == productSpecifications || productSpecifications.Count == 0)
            {
                throw new ArgumentNullException("{0}数据参数不能为空".L10nFormat(nameof(productSpecifications)));
            }
            foreach (var item in productSpecifications)
            {
                var productSpecification = new ProductSpecification();
                productSpecification.ProductId = item.ProductId;
                savedData.Add(productSpecification);
            }

            RF.Save(savedData);
            return true;
        }
    }
}
